R/accounts.R

```diff
@@ -157,83 +157,54 @@ setAccountInfo <- function(name, token, secret, server = "shinyapps.io") {
   check_string(secret)
   check_string(server)
 
+  accountId <- findAccountId(name, token, secret, server)
+
+  registerAccount(
+    serverName = server,
+    accountName = name,
+    accountId = accountId,
+    token = token,
+    secret = secret
+  )
+  invisible()
+}
+
+# A user can have multiple accounts, so iterate over all accounts looking
+# for one with the specified name
+findAccountId <- function(name, token, secret, server) {
   account <- list(token = token, secret = secret, server = server)
   client <- clientForAccount(account)
 
-  # get user Id
   userId <- client$currentUser()$id
 
-  # get account id
   accountId <- NULL
   accounts <- client$accountsForUser(userId)
   for (account in accounts) {
     if (identical(account$name, name)) {
-      accountId <- account$id
-      break
+      return(account$id)
     }
   }
-  if (is.null(accountId))
-    stop("Unable to determine account id for account named '", name, "'")
-
-  registerCloudTokenSecret(
-    serverName = serverInfo$name,
-    accountName = name,
-    userId = userId,
-    token = token,
-    secret = secret,
-  )
-  invisible()
-}
-
-registerCloudTokenSecret <- function(serverName,
-                                     accountName,
-                                     userId,
-                                     accountId,
-                                     token,
-                                     secret) {
-  # get the path to the config file
-  path <- accountConfigFile(accountName, serverName)
-  dir.create(dirname(path), recursive = TRUE, showWarnings = FALSE)
-
-  # write the user info
-  write.dcf(
-    list(
-      name = accountName,
-      userId = userId,
-      accountId = accountName,
-      token = token,
-      secret = secret,
-      server = serverName
-    ),
-    path,
-    width = 100
-  )
-
-  # set restrictive permissions on it if possible
-  if (identical(.Platform$OS.type, "unix"))
-    Sys.chmod(path, mode = "0600")
-
-  path
+  cli::cli_abort("Unable to determine {.arg accountId} for account {.str {name}}")
 }
 
 #' @rdname accounts
 #' @family Account functions
 #' @export
 accountInfo <- function(name = NULL, server = NULL) {
-
   fullAccount <- findAccount(name, server)
   configFile <- accountConfigFile(fullAccount$name, fullAccount$server)
 
   accountDcf <- readDcf(configFile, all = TRUE)
   info <- as.list(accountDcf)
   # remove all whitespace from private key
   if (!is.null(info$private_key)) {
-    info$private_key <- secret(gsub("[[:space:]]", "", info$private_key))
+    info$private_key <- gsub("[[:space:]]", "", info$private_key)
   }
 
-  if (!is.null(info$secret)) {
-    info$secret <- secret(info$secret)
-  }
+  # Hide credentials
+  info$private_key <- secret(info$private_key)
```

Comment:
Seeing this shuffle together makes me appreciate the consistency of `private_key` and `apiKey`. 😏